server/src/main/java/org/apache/druid/client/DirectDruidClient.java

```diff
@@ -551,7 +553,7 @@ private void cancelQuery(Query<T> query, String cancelUrl)
             if (!responseFuture.isDone()) {
               log.error("Error cancelling query[%s]", query);
             }
-            StatusResponseHolder response = responseFuture.get();
+            StatusResponseHolder response = responseFuture.get(5, TimeUnit.SECONDS);
```

Comment:
Should we set it to 30? I don't have a strong opinion but 5 seems low.

Reply:
yes, make sense, let me set to 30 then

Reply:
done